plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/MavenAppEngineFacetUtil.java

```diff
@@ -0,0 +1,145 @@
+    appEngineApiDependency.setScope("");
+    dependencies.add(appEngineApiDependency);
+
+    Dependency servletApiDependency = new Dependency();
+    servletApiDependency.setGroupId("javax.servlet");
+    servletApiDependency.setArtifactId("servlet-api");
+    servletApiDependency.setVersion("2.5");
+    servletApiDependency.setScope("provided");
+    dependencies.add(servletApiDependency);
+
+    Dependency jstlDependecy = new Dependency();
+    jstlDependecy.setGroupId("jstl");
+    jstlDependecy.setArtifactId("jstl");
+    jstlDependecy.setVersion("1.2");
+    dependencies.add(jstlDependecy);
+
+    Dependency appEngineTestingDependency = new Dependency();
+    appEngineTestingDependency.setGroupId("com.google.appengine");
+    appEngineTestingDependency.setArtifactId("appengine-testing");
+    appEngineTestingDependency.setVersion("${appengine.version}");
+    appEngineTestingDependency.setScope("test");
+    dependencies.add(appEngineTestingDependency);
+
+    Dependency appEngineApiStubsDependency = new Dependency();
+    appEngineApiStubsDependency.setGroupId("com.google.appengine");
+    appEngineApiStubsDependency.setArtifactId("appengine-api-stubs");
+    appEngineApiStubsDependency.setVersion("${appengine.version}");
+    appEngineApiStubsDependency.setScope("test");
+    dependencies.add(appEngineApiStubsDependency);
+
+    return dependencies;
+  }
+
+  /**
+   * Returns a map of all the App Engine properties that should exist in the pom.xml
+   * of a maven project that has the App Engine facet installed
+   *
+   * @param monitor to be able to cancel operation
+   * @return a map where the keys and values are the property fields and values respectively
+   */
+  public static Map<String, String> getAppEnginePomProperties(IProgressMonitor monitor) {
+    String appengineArtifactVersion = resolveLatestReleasedArtifact(monitor,
+        "com.google.appengine", "appengine-api-1.0-sdk", "jar", AppEngineStandardFacet.DEFAULT_APPENGINE_SDK_VERSION);
+    String gcloudArtifactVersion = resolveLatestReleasedArtifact(monitor,
+        "com.google.appengine", "gcloud-maven-plugin", "maven-plugin", AppEngineStandardFacet.DEFAULT_GCLOUD_PLUGIN_VERSION);
+
+    Map<String, String> allProperties = new HashMap<String, String>();
+    allProperties.put("app.id", "");
+    allProperties.put("app.version", "1");
+    allProperties.put("appengine.version", appengineArtifactVersion);
+    allProperties.put("gcloud.plugin.version", gcloudArtifactVersion);
+    return allProperties;
+  }
+
+  /**
+   * Returns true if the group IDs and artifact IDs of <code>dependency1</code> and
+   * <@code>dependency2</@code> are equal. Returns false otherwise.
+   */
+  // visible for testing
+  public static boolean areDependenciesEqual(Dependency dependency1, Dependency dependency2) {
+    if ((dependency1 == null) || (dependency2 == null)) {
+      return false;
+    }
+
+    if (dependency1.getGroupId() == null) {
```

Comment:
Ah, you're right. I mixed up the comparisons between the `Dependency` and the `Dependency.getGroupId()` objects.